agent/taskresource/cgroup/cgroup_unsupported.go

```diff
@@ -0,0 +1,148 @@
+// +build !linux
+// Copyright 2014-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cgroup
+
+import (
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/taskresource"
+)
+
+const (
+	resourceName = "cgroup"
+)
+
+// CgroupResource represents Cgroup resource
+type CgroupResource struct {
+	desiredStatusUnsafe taskresource.ResourceStatus
+	knownStatusUnsafe   taskresource.ResourceStatus
+	appliedStatus       taskresource.ResourceStatus
+	// lock is used for fields that are accessed and updated concurrently
+	lock sync.RWMutex
+}
+
+// SetDesiredStatus safely sets the desired status of the resource
+func (c *CgroupResource) SetDesiredStatus(status taskresource.ResourceStatus) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	c.desiredStatusUnsafe = status
+}
+
+// GetDesiredStatus safely returns the desired status of the task
+func (c *CgroupResource) GetDesiredStatus() taskresource.ResourceStatus {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.desiredStatusUnsafe
+}
+
+// GetName safely returns the name of the resource
+func (c *CgroupResource) GetName() string {
+	return resourceName
+}
+
+// DesiredTerminal returns true if the cgroup's desired status is REMOVED
+func (c *CgroupResource) DesiredTerminal() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.desiredStatusUnsafe == taskresource.ResourceRemoved
+}
+
+// KnownCreated returns true if the cgroup's known status is CREATED
+func (c *CgroupResource) KnownCreated() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.knownStatusUnsafe == taskresource.ResourceCreated
+}
+
+// TerminalStatus returns the last transition state of cgroup
+func (c *CgroupResource) TerminalStatus() taskresource.ResourceStatus {
+	return taskresource.ResourceRemoved
+}
+
+// GetNextKnownStateProgression returns the state that the resource should
+// progress to based on its `KnownState`.
+func (c *CgroupResource) GetNextKnownStateProgression() taskresource.ResourceStatus {
+	return c.GetKnownStatus() + 1
+}
+
+// ApplyTransition calls the function required to move to the specified status
+func (c *CgroupResource) ApplyTransition(nextState taskresource.ResourceStatus) (bool, error) {
```

Comment:
All of the methods that return an `error` in this struct should return `unsupported` error.